decrypt_oracle/test/integration/integration_test_utils.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Helper utilities for use by integration tests."""
+import base64
+import json
+import os
+from collections import namedtuple
+
+import pytest
+from aws_encryption_sdk.key_providers.kms import KMSMasterKeyProvider
+
+HERE = os.path.abspath(os.path.dirname(__file__))
+DEPLOYMENT_REGION = "AWS_ENCRYPTION_SDK_PYTHON_DECRYPT_ORACLE_REGION"
+DEPLOYMENT_ID = "AWS_ENCRYPTION_SDK_PYTHON_DECRYPT_ORACLE_API_DEPLOYMENT_ID"
+AWS_KMS_KEY_ID = "AWS_ENCRYPTION_SDK_PYTHON_INTEGRATION_TEST_AWS_KMS_KEY_ID"
+_KMS_MKP = None
+_ENDPOINT = None
+
+
+def decrypt_endpoint():
+    """Build the API endpoint based on environment variables."""
+    global _ENDPOINT  # pylint: disable=global-statement
+
+    if _ENDPOINT is not None:
+        return _ENDPOINT
+
+    try:
+        deployment_id = os.environ[DEPLOYMENT_ID]
+        region = os.environ[DEPLOYMENT_REGION]
+    except KeyError:
+        raise ValueError(
+            (
+                'Environment variables "{region}" and "{deployment}" '
+                "must be set to the correct values for the deployed decrypt oracle."
+            ).format(region=DEPLOYMENT_REGION, deployment=DEPLOYMENT_ID)
+        )
+
+    _ENDPOINT = "https://{deployment_id}.execute-api.{region}.amazonaws.com/api/v0/decrypt".format(
+        deployment_id=deployment_id, region=region
+    )
+    return _ENDPOINT
+
+
+def get_cmk_arn():
+    """Retrieve the target CMK ARN from environment variable."""
+    try:
+        arn = os.environ[AWS_KMS_KEY_ID]
+    except KeyError:
+        raise ValueError(
+            'Environment variable "{}" must be set to a valid KMS CMK ARN for integration tests to run'.format(
+                AWS_KMS_KEY_ID
+            )
+        )
+
+    if arn.startswith("arn:") and ":alias/" not in arn:
+        return arn
+
+    raise ValueError("KMS CMK ARN provided for integration tests must be a key not an alias")
```

Comment:
Can we raise a custom exception type here?

Reply:
Would that add value? The problem that this exception is identifying is that a user-provided value is incorrect. That seems to me to be the definition of `ValueError`.

Reply:
If you expect something to catch this exception, then yes.  Its cheap to provide a custom exception, and we aren't giving up anything that ValueError provides.

Reply:
The only place this error will manifest is in pytest error logs.